nats/client.go

```diff
@@ -130,11 +130,23 @@ func (c *Client) ConsumeSequences(ctx context.Context, handler SequenceHandler)
 		}
 
 		if hopsMsg.MessageId == HopsMessageId {
-			c.logger.Debugf("Skipping hops assignment message")
+			c.logger.Debugf("Skipping 'hops assignment' message")
 
 			err := DoubleAck(ctx, msg)
 			if err != nil {
-				c.logger.Errf(err, "Unable to ack hops assignment message")
+				c.logger.Errf(err, "Unable to ack 'hops assignment' message")
+			}
+
+			return
+		}
+
+		if hopsMsg.Done {
+			// TODO: Actually finalise the pipeline here
+			c.logger.Debugf("Skipping 'pipeline done' message")
+
+			err := DoubleAck(ctx, msg)
```

Comment:
Are you possibly acking again? since the previous if clause also acks.

Reply:
No, previous if clause has a return statement